src/main/java/nwoolcan/model/brewery/production/batch/ModifiableBatchInfoImpl.java

```diff
@@ -0,0 +1,130 @@
+package nwoolcan.model.brewery.production.batch;
+
+import javafx.util.Pair;
+import nwoolcan.model.brewery.production.batch.step.parameter.Parameter;
+import nwoolcan.model.brewery.production.batch.step.parameter.ParameterTypeEnum;
+import nwoolcan.model.brewery.warehouse.article.IngredientArticle;
+import nwoolcan.model.utils.Quantity;
+
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Optional;
+
+//Package-protected
+class ModifiableBatchInfoImpl implements ModifiableBatchInfo {
+    private final BeerDescription beerDescription;
+    private final BatchMethod method;
+    private final Quantity size;
+    private final Collection<Pair<IngredientArticle, Quantity>> ingredients;
+    @Nullable
+    private final WaterMeasurement measurement;
+    @Nullable
+    private Parameter og;
+    @Nullable
+    private Parameter fg;
+    @Nullable
+    private Parameter ebc;
+    @Nullable
+    private Parameter abv;
+    @Nullable
+    private Parameter ibu;
+
+    //Package-private
+    ModifiableBatchInfoImpl(final Collection<Pair<IngredientArticle, Quantity>> ingredients,
```

Comment:
add `@Nullable` to a parameter is a great way to make nullability check inefficient. If I want to call the constructor with a `WaterMeasurement` field i probably meant to pass a valid value, otherwise I call the other constructor. To reuse the code i have to make `watermeasurement` field not final